[#128764711] - Reportar problemas com a vaga

jprmoisakis opened the pull request:
Feature: Report parking space problem
  As a system user
  I want to send a problem report
  So that I can notify the admin about the problem.

Cenários:
Enviar Relatórios de problema.
Erro ao tentar enviar um relatório de problema já existente.

Implementados testes de GUI e controller

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pauloborba/sistemadevagasdeestacionamento/7)
<!-- Reviewable:end -->

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga
> Implementação de Testes de Controlador Features e Steps
> Adição das propriedades "Preferencial" em User e ParkingSpace
> Criação da Classe "ParkingSpaceTestDataAndOperations" com funções auxiliares para objetos ParkingSpace
> Implementação da função Book e UnBook no ParkingSpaceController

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga
> Implementação de cenários e steps de GUI
> Implementação das pages
> Alterações nas views

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga
> correção de conflito

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga
> Resolução de conflito

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga
> Modificação nos steps

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vaga

> Alteração na view de criação de pagina

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vagals

> Remoção de quebras de linha da view

Commit:
[#128764641] - Desenvolver cenários de alteração de reserva de vagals

> Padronização dos metodos de page

joaozenos committed:
[#128764641] - Desenvolver cenários de alteração de reserva de vagals

> Ajustes nos cenários
> Ajustes nos steps de ParkingSpaceBook
> Ajustes na page de visualização

joaozenos committed:
[#128764641] - Desenvolver cenários de alteração de reserva de vagas

> Ajustes nos cenários
> Ajustes nos steps de ParkingSpaceBook
> Ajustes na page de visualização

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> Implementação de Cenários 
> Implementação de testes de gui e controller.
> Atualização de chromedriver para Windows.
> Implementação de views e pages.
> Adicionada nova opção na view HomeIndex.
> Adição de metodo para ser usado nos testes na page Homepage.
> Adição de problemReport na lista de objetos a serem deletados ao fim de cada teste.
> Adicionado tempo de timeout e retryInterval ao executar o chromedriver

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> @ignore removido dos testes implementados previamente

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

>Modificação feita para realização de teste no travis

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> Correção de erro apontado no travis

jets- committed:
modelo e cenarios

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> Adicionada independencia de linguagem

jets- committed:
controller; cenarios controller; view;  alguns dados no bootstrap

jets- committed:
cenario gui (falta terminar e testar); pages;

joaozenos committed:
[#128764641] - Desenvolver cenários de alteração de reserva de vagas

> Adição das seções especificas de desenvolvimento da feature

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> Correção de comportamento dos testes de controller.
> Adição do #if ($ReportParkingSpaceProblem) #end nos trechos de código adicionados ou modificados.

joaozenos committed:
[#128764641] - Desenvolver cenários de alteração de reserva de vagas

> Pequeno ajuste numa view que quebrava um teste.

joaozenos committed:
[FINISHES #128764641] - Desenvolver cenários de alteração de reserva de vagas

> Pequeno ajuste numa view que quebrava um teste.

joaozenos committed:
Merge branch 'master' of https://github.com/joaozenos/SistemaDeVagasDeEstacionamento

jprmoisakis committed:
[FINISHES #128764711]Fim da implementação: Reportar problemas com a vaga

> Atualizando o repositório com o master

jets- committed:
[#133901927] Opção marcar como resolvido problema reportado para vaga

jets- committed:
Merged myfeature into master

[#133901927] Opção para marcar como resolvido problema reportado para vaga

jets- committed:
Merged branch master into master

jets- committed:
[#133901927] Opção para marcar como resolvido problema reportado para vaga

pauloborba commented:
Reviewed 13 of 18 files at r1.
Review status: 13 of 18 files reviewed at latest revision, 4 unresolved discussions.

---

_[test/cucumber/ParkingSpaceSuggestion.feature, line 27 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWy_iYgP5rHfTPzt9BU:-KWy_iYgP5rHfTPzt9BV:b-bquy49) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/1a88a1b9f61e146a423f70f16ea9b59fc9134e50/test/cucumber/ParkingSpaceSuggestion.feature#L27)):_

> ``` Cucumber
> 
>   Scenario: The system does not have a parking space available on the desired sector
>     Given the system has stored the user "pmmc" with password "101112" and preference for parking spaces in the "Área II" sector
> ```

por que tirou as passwords?

---

_[test/cucumber/ReportParkingSpaceProblem.feature, line 26 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaOIYEnMDhnFPvWAZ:-KWyaOIYEnMDhnFPvWA_:b-tv1jvs) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L26)):_

> ``` Cucumber
>   When I go to report parking space problem page
>   And I fill the report informations with Title "Carro mal estacionado", Sector "Cin" and Description "Carro de placa XXX -1111"
>   And I send the problem report
> ```

poderia juntar os dois passos. abstrair mais. olha o livro

---

_[test/cucumber/ReportParkingSpaceProblem.feature, line 32 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaBqd-2tqDGzoAUGO:-KWyaBqd-2tqDGzoAUGP:b-90bbx3) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L32)):_

> ``` Cucumber
>   Given I signed up in the system as "joao" with "CIn" as preferred sector
>   When I go to report parking space problem page
>   And I fill the report informations with Title "irregularidade na vaga E5", Sector "Cin" and Description "Carro de placa XXX -1111"
> ```

deveria ser no given

---

_[test/cucumber/steps/ReportParkingSpaceProblemSteps.groovy, line 15 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaa9n7B9V4ssgr433:-KWyaa9n7B9V4ssgr434:b-bn95aj) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/steps/ReportParkingSpaceProblemSteps.groovy#L15)):_

> ``` Groovy
> Given(~/^the system has stored the user "([^"]*)" with "([^"]*)" as preferred sector$/) { String username, String sector ->
>     AuthHelper.instance.signup(username, sector)
>     def user = User.findByUsername(username)
> ```

deveria criar o usuário, não procurar por ele. como isso vai passar?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7)_

<!-- Sent from Reviewable.io -->

jprmoisakis committed:
[#128764711] - Criar cenários para reportar problemas com a vaga.

> Correção de cenários após Feedback

pauloborba commented:
Review status: 13 of 18 files reviewed at latest revision, 5 unresolved discussions.

---

*[test/cucumber/ReportParkingSpaceProblem.feature, line 1 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KXGNYCtpVRiI1bfxSH5:-KXGNYCtpVRiI1bfxSH6:b-570z3q) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L1)):*
> ```Cucumber
> #++ Reportar problemas com a vaga (vaga danificada ou uso incorreto).
> ```

criar dois cenários a mais para refinar a funcionalidade

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7)*
<!-- Sent from Reviewable.io -->

jprmoisakis committed:
[FINISHES #128764711]Fim da implementação: Reportar problemas com a vaga

> Criação  e implementação de dois cenários adicionais

Commit:
[FINISHES #128764641] - Desenvolver cenários de alteração de reserva de vagas

jprmoisakis commented:
Review status: 13 of 18 files reviewed at latest revision, 5 unresolved discussions.

---

*[test/cucumber/ParkingSpaceSuggestion.feature, line 27 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWy_iYgP5rHfTPzt9BU:-KXGReGKtf106vvmLdwf:b-colmc) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/1a88a1b9f61e146a423f70f16ea9b59fc9134e50/test/cucumber/ParkingSpaceSuggestion.feature#L27)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary>
> por que tirou as passwords?
</details>
Horácio retirou no repositório principal.

---

*[test/cucumber/ReportParkingSpaceProblem.feature, line 1 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KXGNYCtpVRiI1bfxSH5:-KXGRLQYKdqJfiEgHZZ8:b-896fix) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L1)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary>
> criar dois cenários a mais para refinar a funcionalidade
</details>
Done.

---

*[test/cucumber/ReportParkingSpaceProblem.feature, line 26 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaOIYEnMDhnFPvWAZ:-KXGPLXA09z3NV18o_tm:b-896fix) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L26)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary>
> poderia juntar os dois passos. abstrair mais. olha o livro
</details>
Done.

---

*[test/cucumber/ReportParkingSpaceProblem.feature, line 32 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaBqd-2tqDGzoAUGO:-KXGRUdgyOP9acQIgWAV:b-896fix) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/ReportParkingSpaceProblem.feature#L32)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary>
> deveria ser no given
</details>
Done.

---

*[test/cucumber/steps/ReportParkingSpaceProblemSteps.groovy, line 15 at r1](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7#-KWyaa9n7B9V4ssgr433:-KXGRsRTHfd6CYXUTeLF:b-eljmy3) ([raw file](https://github.com/pauloborba/sistemadevagasdeestacionamento/blob/23810a7dccbce2dcb29a9fee44ad6d5f57d70379/test/cucumber/steps/ReportParkingSpaceProblemSteps.groovy#L15)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary>
> deveria criar o usuário, não procurar por ele. como isso vai passar?
</details>
"AuthHelper.instance.signup(username, sector)" cria o usuário, a linha seguinte serve para verificar se o usuário foi criado com sucesso. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/sistemadevagasdeestacionamento/7)*
<!-- Sent from Reviewable.io -->

Commit:
[FINISHES #128764641] - Desenvolver cenários de alteração de reserva de vaga

jets- committed:
[#133901927] controllers funcionando

> corrigido cenários de controller
> nova versao do chromedriver

jets- committed:
[#133901927] Cenários de gui funcionando

> corrigido erro de usuario nao autorizado
> troca de elemento botao para corrigir
> todos os 20 cenarios funcionando

jets- committed:
[#133901927] Internacionalização e usuario

> Internacionalização
> organização de codigo
> ajustes de gui e controller para a internacionalização
> respostas para o usuario

jets- committed:
[FINISHES #133901927] Marcar como resolvido problema reportado para vaga

jprmoisakis committed:
Merge dos pull requests

> Resolução de conflitos na homepage
> Resolução de conflitos nas pages

jprmoisakis committed:
Ajustes pós merge

> Ajuste de parametro ao fazer signup

jprmoisakis committed:
Merge remote-tracking branch 'refs/remotes/jets/master'

Conflicts:
	grails-app/controllers/sistemadevagasdeestacionamento/ProblemReportController.groovy
	grails-app/domain/sistemadevagasdeestacionamento/ProblemReport.groovy
	grails-app/views/home/index.gsp
	grails-app/views/problemReport/index.gsp
	test/functional/pages/HomePage.groovy

jprmoisakis committed:
Adição de Método auxiliar

> Faz o problem report poder ser marcado como "resolvido" a nível de controlller.

jprmoisakis committed:
Conflitos de internacionalização resolvidos

jprmoisakis committed:
Minor fixes

jprmoisakis committed:
Adição de gui

> Adicionada a gui para a feature de marcar como resolvido o relatório de problema

jprmoisakis committed:
Ajustes nos Testes

> Retirada de método duplicado
> Retirada de método na pagina de testes
> Ajuste no arquivo feature, estava sem Then

jprmoisakis committed:
Fim das correções dos testes da nova Feature

> Ajuste no arquivo ParkingSpaceSolvedReport.feature, passo não fazia nada
> Correção de passo de teste de gui estava sendo feito via controller
> Adição de ProblemReportShowPage
